spec/index.bs

```diff
@@ -114,33 +168,131 @@ interface GPUAdapter : GPUObjectBase {
 
-<script type=idl>
-[Exposed=Window]
-interface GPU {
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
-};
-</script>
+To get a {{GPUAdapter}}, use {{GPU/requestAdapter()}}.
 
 <script type=idl>
 dictionary GPURequestAdapterOptions {
-    GPUPowerPreference? powerPreference;
+    GPUPowerPreference powerPreference;
 };
 </script>
 
+<dfn dictionary>GPURequestAdapterOptions</dfn>
+provides requirements and hints to the user agent indicating what
+configuration is suitable for the application.
+
+<dl dfn-type=dict-member dfn-for=GPURequestAdapterOptions>
+    : <dfn>powerPreference</dfn>
+    ::
+        Provides a hint indicating what class of [=adapter=] should be selected from the system's
+        available adapters.
+
+        If unspecified, the user agent decides which class of adapter is most suitable.
+
+        Note:
+        This hint may influence which GPU is used in a system with multiple GPUs.
+        For example, a dual-GPU system might have one GPU that consumes less power at the expense
+        of rendering performance.
+
+        Note:
+        Depending on the exact hardware configuration, such as attached displays or removable
+        GPUs, the user agent may select different [=adapters=] given the same power preference.
+        Typically, given the same hardware configuration and state (e.g. battery percentage) and
+        `powerPreference`, the user agent should select the same adapter.
+</dl>
+
 <script type=idl>
 enum GPUPowerPreference {
     "low-power",
     "high-performance"
 };
 </script>
 
-## GPUDevice ## {#device}
+<dl dfn-type=enum-value dfn-for=GPUPowerPreference>
+    : <dfn>"low-power"</dfn>
+    ::
+        Indicates a request to prioritize power savings over rendering performance.
+
+        Note:
+        Generally, content should use this if it is unlikely to be constrained by drawing
+        performance; for example, if it renders only one frame per second, draws only relatively
+        simple geometry with simple shaders, or uses a small HTML canvas element.
+        Developers are encouraged to use this value if their content allows, since it may
+        significantly improve battery life on portable devices.
+
+    : <dfn>"high-performance"</dfn>
+    ::
+        Indicates a request to prioritize rendering performance over power consumption.
+
+        Note:
+        By choosing this value, developers should be aware that, for [=devices=] created on the
+        resulting adapter, user agents are more likely to force device loss, in order to save
+        power.
+        Developers are encouraged to only specify this value if they believe it is absolutely
+        necessary, since it may significantly decrease battery life on portable devices.
+</dl>
+
+## Devices ## {#devices}
+
+A <dfn dfn>device</dfn> is the basic "context" through which [=internal objects=] are created.
```

Comment:
I'm not sure if "context" should be used here, it's quite abstract. Perhaps, "is the logical instantiation of an adapter" would do better?